pkg/installmanager/installmanager.go

```diff
@@ -223,6 +224,23 @@ func (m *InstallManager) Run() error {
 		os.Exit(0)
 	}
 
+	// will be non-nil if we've been unable to setup SSH, which is a non-fatal error:
+	var sshAgentSetupErr error
+
+	var sshKeyPath string
+	sshKeyPath, sshAgentSetupErr = m.initSSHKey()
+	if sshAgentSetupErr != nil {
+		// Not a fatal error.
+		m.log.WithError(sshAgentSetupErr).Info("unable to initialize ssh agent")
+	} else {
+		sshCleanupFunc, sshAgentSetupErr := m.initSSHAgent(sshKeyPath)
+		defer sshCleanupFunc()
+		if sshAgentSetupErr != nil {
+			// Not a fatal error.
+			m.log.WithError(sshAgentSetupErr).Info("ssh agent is not initialized")
+		}
+	}
```

Comment:
Moving SSH agent setup code here, we are assuming the SSH private key can also be used to hit libvirt over ssh as I didn't want to add another SSH key to the API, as this is also not likely permanent. We now always initialize SSH if we can, and if we cannot it's not a fatal error.

I will run a test that log gathering still works and report back.

Reply:
Should this be an error if it is a bare-metal install? Or is it OK to wait until the later failure to error out if we cannot initialize SSH?

Reply:
Yeah it will quickly error elsewhere if SSH isn't available, so this form leaves us a little closer to long term form.

Reply:
(the SSH need won't be permanent I should have mentioned)